### DISCUSSION TITLE
Using the AWS Secrets Manager data source

### DISCUSSION THREAD

**Discussion by sjiveson** [hairyhenderson/gomplate#2203] [Q&A] [answered]
Hi,

This is a great tool so thank you for making it. 

I'm attempting to render some templates and populate some values with secrets from AWS Secrets Manager. I'm currently unable to use v4.x due to this issue: https://github.com/hairyhenderson/gomplate/issues/2190 so I'm using v3.11.8. I'm able to pull secrets just fine, even from other AWS accounts so all good there. I am however, doing so in a suboptimal way when I need to pull secrets from more than one Secrets Manager path. What I'd like to do is specify Secrets Manager (no URI) as a data source and then specify the URI in the template like so:

```
# Configuration file (.gomplate.yaml):
datasources:
  SM:
    url: aws+sm:

# Template (test.tmpl):
The secret is: {{ (ds "SM" "gomplate/awssm/test") }}

# Command:
./gomplate --file test.tmpl --out test.txt
```
But this fails with an error: `error="yaml: line 3: mapping values are not allowed in this context"`. Adding `///` to the datasource URL does not help. This is despite this working using stdin:
```
echo '{{ (ds "SM" "gomplate/awssm/test") }}' | ./gomplate --verbose -d SM=aws+sm:
```

This does work:
```
# Configuration file (.gomplate.yaml):
datasources:
  SM:
    url: aws+sm:gomplate/awssm/test

# Template (test.tmpl):
The secret is: {{ (ds "SM") }}

# Command:
./gomplate --file test.tmpl --out test.txt
```
But I'd rather not have to configure a dedicated datasource for every secret I'd like to retrieve.

Very likely there's something I'm failing to understand here. If anyone has any pointers on what I may be doing wrong, I'd appreciate your input. Thanks

**Comment by hairyhenderson** [comment, marked as answer]
> But this fails with an error: error="yaml: line 3: mapping values are not allowed in this context".

I think that's just a YAML syntax error. If you surround `aws+sm:` in quotes it'll probably work:

```yaml
# Configuration file (.gomplate.yaml):
datasources:
  SM:
    url: 'aws+sm:'
```

**Reply by sjiveson** [reply]
Thanks for your understanding and time.
